lib/config/fileconf.go

```diff
@@ -855,6 +855,9 @@ type PAM struct {
 	// UsePAMAuth specifies whether to trigger the "auth" PAM modules from the
 	// policy.
 	UsePAMAuth bool `yaml:"use_pam_auth"`
+
+	// Override specifies if local configuration will override cluster PAM configuration.
+	Override bool `yaml:"override"`
```

Comment:
Do we need an additional field for override? If the user has specified local configuration, that feels like it shoudl be sufficient to override cluster wide configuration.

For example, if a user specifies a command line flag it usually overrides file configuration, the user does not have to specify `--override` as well.

Reply:
Well we need a way to tell if the user supplied any configuration vs did not supply configuration and this was the most intuitive way to do it IMO. The other option is refactoring our configuration system to allow us to figure that out in some other way.

Reply:
But changing fundamentally how our configuration system shows this feels out of scope for this PR nor do I think I have time for it and changing behaviour for just the PAM config feels awfully inconsistent.

Reply:
File configuration get transformed into `*pam.Config` in the `service` package: https://github.com/gravitational/teleport/blob/master/lib/service/cfg.go#L499.

You can just check if it's `nil` or not to determine if the user set it.

Reply:
@xacrimon @russjones Andrejs has solved the problem of static vs dynamic configuration:

https://github.com/gravitational/teleport/blob/master/rfd/0016-dynamic-configuration.md

I would avoid introducing a new configuration parameter if possible, and with Andrej's design it seems quite possible.